tools/depcheck/glide/glide.go

```diff
@@ -0,0 +1,46 @@
+package glide
+
+import (
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+type LockFile struct {
+	Hash    string    `yaml:"hash"`
+	Updated time.Time `yaml:"updated"`
+
+	Imports []*LockFileImport `yaml:"imports"`
+}
+
+func (l *LockFile) Decode(b []byte) error {
+	return yaml.Unmarshal(b, l)
+}
+
+type YamlFile struct {
+	Package     string             `yaml:"package"`
+	ExcludeDirs []string           `yaml:"excludeDirs"`
+	Imports     YamlFileImportList `yaml:"import"`
+}
+
+func (y *YamlFile) Decode(b []byte) error {
+	return yaml.Unmarshal(b, y)
+}
+
+type LockFileImport struct {
+	Name    string `yaml:"name"`
+	Repo    string `yaml:"repo,omitempty"`
+	Version string `yaml:"version"`
+}
+
+type YamlFileImport struct {
+	Package string `yaml:"package"`
+	Repo    string `yaml:"repo,omitempty"`
+	Version string `yaml:"version"`
+}
+
+type YamlFileImportList []*YamlFileImport
+
+func (l *YamlFileImportList) Encode() ([]byte, error) {
```

Comment:
Why?  Avoid overriding the encoding decoding unless you have a compelling reason.

Reply:
I originally added this method here so that other packages would not have to know about `gopkg.in/yaml.v2` - so in `PinImportOpts.Run`, when I receive a YamlFileImportList that is created in `MissingImports`, all I have to do is call `Encode` on the result without worrying about which encoder to use

Reply:
> I originally added this method here so that other packages would not have to know about gopkg.in/yaml.v2 - so in PinImportOpts.Run, when I receive a YamlFileImportList that is created in MissingImports, all I have to do is call Encode on the result without worrying about which encoder to use

I'll merge it, but it seems like encoding should be a choice made the caller, not by the callee.  This appears overly restrictive.